src/uct/api/uct.h

```diff
@@ -184,9 +184,12 @@ enum {
     UCT_IFACE_FLAG_CONNECT_TO_IFACE = UCS_BIT(40), /**< Supports connecting to interface */
     UCT_IFACE_FLAG_CONNECT_TO_EP    = UCS_BIT(41), /**< Supports connecting to specific endpoint */
 
-    /* Thread safety */
-    UCT_IFACE_FLAG_AM_THREAD_SINGLE = UCS_BIT(44), /**< Active messages callback is invoked only from main thread */
-
+    /* active message callback invocation */
+    UCT_IFACE_FLAG_AM_CB_SYNC       = UCS_BIT(44), /**< Active messages callback is invoked only when
+                                                        uct_worker_progress() is called */
+    UCT_IFACE_FLAG_AM_CB_ASYNC      = UCS_BIT(45), /**< Active message callback can be invoked at any
```

Comment:
"callbacks installed with CB_ASYNC flags would be invoked within a reasonable amount of time even if uct_worker_progress is not being called, possibly from another thread."